plugins/bulk-import-backend/src/service/router.test.ts

```diff
@@ -132,919 +107,529 @@ const configuration = new ConfigReader({
+      getEntityByRef: jest.fn(),
+      removeEntityByUid: jest.fn(),
       refreshEntity: jest.fn(),
+      getEntityFacets: jest.fn(),
+      getLocationById: jest.fn(),
+      getLocationByRef: jest.fn(),
+      addLocation: jest.fn(),
+      removeLocationById: jest.fn(),
+      getLocationByEntity: jest.fn(),
+      validateEntity: jest.fn(),
     } as unknown as CatalogClient;
-    const voidLogger = getVoidLogger();
-    mockCatalogInfoGenerator = new CatalogInfoGenerator(
-      voidLogger,
-      mockDiscovery,
-      mockAuth,
-      mockCatalogClient,
-    );
-    mockGithubApiService = new GithubApiService(
-      voidLogger,
-      configuration,
-      mockCache,
-    );
-    const router = await createRouter({
-      logger: voidLogger,
-      config: configuration,
-      permissions: permissionEvaluator,
-      cache: mockCache,
-      discovery: mockDiscovery,
-      catalogApi: mockCatalogClient,
-      identity: mockIdentityClient,
-      catalogInfoHelper: mockCatalogInfoGenerator,
-      githubApi: mockGithubApiService,
-    });
-    app = express().use(router);
   });
 
-  beforeEach(() => {
+  afterEach(() => {
     jest.resetAllMocks();
     jest.restoreAllMocks();
+    server.resetHandlers();
   });
 
+  function addHandlersForGHTokenAppErrors() {
+    server.use(
+      rest.get(`${LOCAL_ADDR}/user/orgs`, (_, res, ctx) =>
+        res(
+          ctx.status(500),
+          ctx.json({ message: 'Github Token auth did not succeed' }),
+        ),
+      ),
+      rest.get(`${LOCAL_ADDR}/app/installations`, (_, res, ctx) =>
+        res(
+          ctx.status(403),
+          ctx.json({ message: 'Github App auth returned an error' }),
+        ),
+      ),
+      rest.get(`${LOCAL_ADDR}/user/repos`, (_, res, ctx) =>
+        res(
+          ctx.status(401),
+          ctx.json({ message: 'Github Token auth did not succeed' }),
+        ),
+      ),
+      rest.get(`${LOCAL_ADDR}/orgs/some-org/repos`, (_, res, ctx) =>
+        res(
+          ctx.status(502),
+          ctx.json({ message: 'Github Token auth did not succeed' }),
+        ),
+      ),
+    );
+  }
+
+  async function startNewBackendServer(
+    authorizeResult?: AuthorizeResult.DENY | AuthorizeResult.ALLOW,
+    config?: any,
+  ) {
+    const features: (BackendFeature | Promise<{ default: BackendFeature }>)[] =
+      [
+        bulkImportPlugin,
+        mockServices.rootLogger.factory,
+        mockServices.rootConfig.factory({
+          data: { ...BASE_CONFIG, ...(config || {}) },
+        }),
+        mockServices.cache.factory,
+        createServiceFactory({
+          service: catalogServiceRef,
+          deps: {},
+          factory: () => mockCatalogClient,
+        }),
+      ];
+    if (authorizeResult) {
+      features.push(
+        mockServices.permissions.mock({
+          authorize: async () => [{ result: authorizeResult }],
+        }).factory,
+      );
+    }
+    return startTestBackend({ features }).then(backend => backend.server);
```

Comment:
```suggestion
    return (await startTestBackend({ features })).server;
```
Please use async/await rather than callbacks.